### ISSUE TITLE
Model import fails for keras2 model when Merge layer exists

### ISSUE THREAD

**Issue by kullanici0606** [deeplearning4j/deeplearning4j#4552]
#### Importing keras2 model that has a Merge layer fails with NullPointerException.
I am trying to import my keras model which has a Merge in its first layer in the Sequential model, however it fails with NPE. Roughly the code is as follows:
`model1= ... # A sequential model with some layers in it`
`model2= ... # Another sequential model with some layers in it`
`final = Sequential()`
`final.add(merged)`
`final.add(Dense(1))`
`final.add(Activation('sigmoid'))`

It seems that import code is trying to find the input_shape but since Merge layer does not have one, it returns null when asked and rest of the process fails with NPE. I am attaching the example keras code, model.json and model weights. [code_and_exported_model.zip](https://github.com/deeplearning4j/deeplearning4j/files/1662983/code_and_exported_model.zip)

I tried the import the model with following code:

`MultiLayerNetwork network = KerasModelImport.importKerasSequentialModelAndWeights("model_with_merge.json", "model_with_merge_weights.h5", false);`

Here is the relevant stack trace:

`Exception in thread "main" java.lang.NullPointerException`
`	at org.deeplearning4j.nn.modelimport.keras.layers.KerasInput.<init>(KerasInput.java:82)`
`	at org.deeplearning4j.nn.modelimport.keras.layers.KerasInput.<init>(KerasInput.java:62)`
`	at org.deeplearning4j.nn.modelimport.keras.KerasSequentialModel.<init>(KerasSequentialModel.java:107)`
`	at org.deeplearning4j.nn.modelimport.keras.KerasSequentialModel.<init>(KerasSequentialModel.java:60)`
`	at org.deeplearning4j.nn.modelimport.keras.utils.KerasModelBuilder.buildSequential(KerasModelBuilder.java:128)`
`	at org.deeplearning4j.nn.modelimport.keras.KerasModelImport.importKerasSequentialModelAndWeights(KerasModelImport.java:218)`

### Expected behavior:
Should import the model

### Encountered behavior
Fails with NullPointerException because of input_shape detection mechanism.

#### Version Information
I used the master branch with this commit: 1f715a2d4d7714fcba05c81631a72f4d4a84261a
0.9.2-SNAPSHOT

* deeplearning4j version 0.9.2-SNAPSHOT
* Keras version: 2.1.1
* Tensorflow version 1.4.0
* Windows 10
* No CUDA
* No NVIDIA driver

**Comment by maxpumperla** [comment]
@kullanici0606 thanks for reporting, that's indeed a problem. For now you could use the functional API, i.e. something like [this](https://gist.github.com/emc5ud/3fd5d22a63bc65e918083a3eff3fdcad), which does work.

**Comment by kullanici0606** [comment]
@maxpumperla Thanks for help and example. I will try it.

**Comment by maxpumperla** [comment]
@kullanici0606 just a quick update on this: it's more complicated then anticipated. `Merge` has been deprecated for a long time and removed by now. `Merge` holds two complete `Sequential` models in its _layer_ config, which we have to recursively read and make sense of.

It's much easier to map this with the functional API. I'll get back to this at one point, but it will take more itme.

**Comment by kullanici0606** [comment]
I think I've managed to import Keras model after using functional API as you suggested. Therefore this is not a blocking issue for us. I can close it if you also prefer so, since as you said, merge is deprecated.

Thanks.
